registry/client/transport/http_reader.go

```diff
@@ -0,0 +1,167 @@
+package transport
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/docker/distribution"
```

Comment:
@stevvooe i wonder whether the `transport` package should have a dependency on distribution or if it should just recreate the ReadSeekCloser interface